gdg/__init__.py

```diff
@@ -238,7 +276,35 @@ def list(self, gallery=""):
             images = images.order_by(SQL('path collate nocase'))
             result["images"] = [get_relative_path(baseurl, i.path) for i in images]
         return result
-    
+
+    @cherrypy.expose
+    @cherrypy.tools.allow(methods=['PUT', 'POST'])
+    def tag(self, **kwargs):
+        if not verify_key(kwargs.get('key', '')):
+            return "What's the magic word?"
+        tag_name = kwargs.get('tag', '')
```

Comment:
`tag` and `file` should be function parameters so they can be passed in the request body instead of the URL.